src/microbe_stage/Microbe.Contact.cs

```diff
@@ -430,18 +579,153 @@ public void SendOrganellePositionsToMembrane()
+
+        // TODO: once the colony leader can leave without the entire colony disbanding this perhaps should keep the
+        // disband entire colony functionality
+        Colony!.RemoveFromColony(this);
+    }
+
+    internal void OnColonyMemberRemoved(Microbe microbe)
+    {
+        cachedColonyRotationMultiplier = null;
+
+        if (microbe == this)
+        {
+            OnUnbound?.Invoke(this);
+
+            RevertNodeParent();
+            ai?.ResetAI();
+
+            Mode = ModeEnum.Rigid;
+
+            return;
+        }
+
+        if (IsMulticellular && Colony?.Master == this)
+        {
+            // Lost a member of the multicellular organism
+            OnMulticellularColonyCellLost(microbe);
+        }
+
+        if (HostileEngulfer != microbe)
+            microbe.RemoveCollisionExceptionWith(this);
+        if (microbe.HostileEngulfer != this)
+            RemoveCollisionExceptionWith(microbe);
+    }
+
+    internal void ReParentShapes(Microbe to, Vector3 offset)
+    {
+        // TODO: if microbeRotation is the rotation of *this* instance we should use the variable here directly
+        // An object doesn't need to be told its own member variable in a method...
+        // https://github.com/Revolutionary-Games/Thrive/issues/2504
+        foreach (var organelle in organelles!)
+            organelle.ReParentShapes(to, offset);
+    }
+
+    internal void OnColonyMemberAdded(Microbe microbe)
+    {
+        cachedColonyRotationMultiplier = null;
+
+        if (microbe == this)
+        {
+            if (Colony == null)
+            {
+                throw new InvalidOperationException(
+                    $"{nameof(Colony)} must be set before calling {nameof(OnColonyMemberAdded)}");
+            }
+
+            OnIGotAddedToColony();
+
+            if (Colony.Master != this)
+            {
+                Mode = ModeEnum.Static;
+            }
+
+            ReParentShapes(Colony.Master, GetOffsetRelativeToMaster());
+        }
+        else
+        {
+            AddCollisionExceptionWith(microbe);
+            microbe.AddCollisionExceptionWith(this);
+        }
+    }
+
+    internal void SuccessfulScavenge()
+    {
+        GameWorld.AlterSpeciesPopulation(Species,
+            Constants.CREATURE_SCAVENGE_POPULATION_GAIN,
+            TranslationServer.Translate("SUCCESSFUL_SCAVENGE"));
+    }
+
+    internal void SuccessfulKill()
+    {
+        GameWorld.AlterSpeciesPopulation(Species,
+            Constants.CREATURE_KILL_POPULATION_GAIN,
+            TranslationServer.Translate("SUCCESSFUL_KILL"));
+    }
+
+    /// <summary>
+    ///   Operations that should be done when this cell is killed. ONLY use this independently of <see cref="Kill"/>
+    ///   if you've already made sure that this microbe is marked as dead since this doesn't do that.
+    /// </summary>
+    /// <returns>
+    ///   The dropped corpse chunks.
+    /// </returns>
+    private HashSet<FloatingChunk> OnKilled()
```

Comment:
Shouldn't this return just a list or IEnumerable? I don't think hashset is really needed here, as this has 0% chance of creating duplicate chunk objects anyway.